### PR TITLE
Update ScoreBoard.hx

### PR DESCRIPTION
**Pull request by SeiferTim**
Add an optional Callback to `postScore` to know when posting is completed successfully. (Otherwise, you might miss this new score when grabbing the hiscores)

### CODE REVIEW

**Comment by SeiferTim** [comment]
If there is a better way to do this, then you can scrap this PR - this is just how I managed to get it to work for me.

File: lib/Source/io/newgrounds/objects/ScoreBoard.hx [hunk 1]
```diff
@@ -65,15 +65,16 @@ class ScoreBoard extends Object<RawScoreBoardData> {
...
 	
-	public function postScore(value :Int, tag:String = null):Void {
-		
-		_core.calls.scoreBoard.postScore(id, value, tag)
-			.send();
+	public function postScore(value:Int, tag:String = null, ?callback:Void->Void = null):Void {
+		if (callback != null)
+			_core.calls.scoreBoard.postScore(id, value, tag).addSuccessHandler(callback).send();
+		else
+			_core.calls.scoreBoard.postScore(id, value, tag).send();
```
Comment (Geokureli) [inline lib/Source/io/newgrounds/objects/ScoreBoard.hx:72]:
```suggestion
	public function postScore(value:Int, tag:String = null, ?callback:(Outcome<CallError>)->Void):Void {
		var call = _core.calls.scoreBoard.postScore(id, value, tag);
		
		if (callback != null)
			call.addOutcomeHandler(callback);
		
		call.send();
```
Note: I didn't test this

**Review by Geokureli — CHANGES_REQUESTED** [review]
This will only call the callback on success, the API uses outcomes for callbacks so that:

1. The callback is called on failure as well as success
2. The dev knows why it failed, if it does (I.E.: network issue, general server issues, or issues with the specific api call)

Look into the callback for [requestScores](https://github.com/Geokureli/Newgrounds/blob/master/lib/Source/io/newgrounds/objects/ScoreBoard.hx#L44-L66). I've also provided an untested suggestion

File: lib/Source/io/newgrounds/objects/ScoreBoard.hx (hunk 1 above)
Comment (SeiferTim) [reply lib/Source/io/newgrounds/objects/ScoreBoard.hx:72]:
```suggestion
	public function postScore(value:Int, tag:String = null, ?callback:Null<(CallOutcome<PostScoreData>) -> Void>):Void {
		final call = _core.calls.scoreBoard.postScore(id, value, tag);
		if (callback != null)
			call.addOutcomeHandler(callback);
		call.send();
	}
```
This works.

**Review by SeiferTim — COMMENTED** [review]
I tested this code and it works. (I don't understand these reviews lol)

**Review by Geokureli — APPROVED** [review]

**Comment by Geokureli** [comment]
I see what I did wrong, sorry 1 sec

File: lib/Source/io/newgrounds/objects/ScoreBoard.hx [hunk 2]
```diff
@@ -65,15 +65,26 @@ class ScoreBoard extends Object<RawScoreBoardData> {
...
-	public function postScore(value :Int, tag:String = null):Void {
+	public function postScore(value:Int, tag:String = null, ?callback:(Outcome<CallError>)->Void):Void {
+		final call = _core.calls.scoreBoard.postScore(id, value, tag);
 		
-		_core.calls.scoreBoard.postScore(id, value, tag)
-			.send();
+		if (callback != null) {
+			
+			call.addOutcomeHandler(
+				(o)->switch (o) {
```
Comment (Geokureli) [inline lib/Source/io/newgrounds/objects/ScoreBoard.hx:74]: FYI `(o)->switch (o) {` is just shorthand for

```hx
function (outcome:CallOutcome<PostScoreData>) {
	
	switch (o) {
		//...
	}
}
```

**Comment by Geokureli** [comment]
I tested this, it compiles, but I need to use it in a test project, while I do that would you be able to test your current use case against this?

**Comment by Geokureli** [comment]
thanks!

**Pull request merged**